pynamodb/expressions/operand.py

```diff
@@ -298,9 +298,12 @@ def __getitem__(self, item: Union[int, str]) -> 'Path':
     def __or__(self, other):
         return _IfNotExists(self, self._to_operand(other))
 
-    def set(self, value: Any) -> SetAction:
+    def set(self, value: Any) -> Union[SetAction, RemoveAction]:
         # Returns an update action that sets this attribute to the given value
```

Comment:
```suggestion
        # Returns an update action that sets this attribute to the given value.
        # For attributes that may not be empty (e.g. sets), this may result
        # in a remove action.
```